src/shared/components/common/markdown-textarea.tsx

```diff
@@ -91,26 +89,25 @@ export class MarkdownTextArea extends Component<
 
     this.handleLanguageChange = this.handleLanguageChange.bind(this);
     this.handleEmoji = this.handleEmoji.bind(this);
-
-    if (isBrowser()) {
-      this.tribute = setupTribute();
-    }
   }
 
-  componentDidMount() {
-    const textarea: any = document.getElementById(this.id);
-    if (textarea) {
-      autosize(textarea);
-      this.tribute.attach(textarea);
-      textarea.addEventListener("tribute-replaced", () => {
-        this.setState({ content: textarea.value });
-        autosize.update(textarea);
-      });
-
-      this.quoteInsert();
-
-      if (this.props.focus) {
-        textarea.focus();
+  async componentDidMount() {
+    if (isBrowser()) {
+      const tribute = await setupTribute();
+      const textarea: any = document.getElementById(this.id);
+      if (textarea) {
+        autosize(textarea);
+        tribute.attach(textarea);
+        textarea.addEventListener("tribute-replaced", () => {
+          this.setState({ content: textarea.value });
+          autosize.update(textarea);
+        });
+
+        this.quoteInsert();
+
+        if (this.props.focus) {
+          textarea.focus();
+        }
```

Comment:
I like this a lot more than how tributes are currently handled.

Reply:
It works but I'm also a bit scared that because this isn't a global import anymore, it might be doing a lot of work to re-import tribute for every single comment.

Reply:
You might be able to cache it in memory by making it a module scoped variable in `markdown.ts`. You can make it `Translation | undefined` (you can import only types with `import { default as TranslationType } from "tributejs"`, which shouldn't cause issues at runtime), check if it already exists in `setupTribute`, and only do the dynamic import if it doesn't exist.

Reply:
I updated it now, I wasn't able to type it because it uses a lot of generics unfortunately. But I added a console debug message, and it seems to only run once, and correctly.